bin/remote-prover/src/lib.rs

```diff
@@ -1,6 +1,41 @@
-pub mod api;
-pub mod error;
-pub mod generated;
+//! Miden Remote Prover
+//!
+//! This crate provides both client and server functionality for Miden's remote proving service.
+//!
+//! ## Features
+//!
+//! - `client`: Enable client functionality (all remote provers: transaction, batch, block)
+//! - `server`: Enable server functionality (binary and library)
+//! - `std`: Enable standard library support
+//! - `concurrent`: Enable concurrent execution
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(not(feature = "std"), feature = "client"))]
+extern crate alloc;
 
-/// Component identifier for structured logging and tracing
+// Component identifier for structured logging and tracing
 pub const COMPONENT: &str = "miden-remote-prover";
+
+// Conditional module exports based on features
+#[cfg(feature = "client")]
+pub mod client;
+
+#[cfg(feature = "server")]
+pub mod server;
+
+pub mod generated;
+pub mod shared;
```

Comment:
A few thoughts on the organization:

- Why do we need the `server` module to be a part of the library? Could we not just import it directly in `main.rs` file? Or is the intent for the server component to be accessible programmatically - i.e., out side of the binary?
- If we can get rid of the `server` module, the library could just export the client structs and there is no need for the separate `client` module.

I guess the only thing that we may want to share is the `generated` module - but even then, we could probably keep them separate.